gms/gossiper.cc

```diff
@@ -1285,6 +1285,20 @@ future<> gossiper::do_gossip_to_unreachable_member(gossip_digest_syn message) {
                     addrs.insert(x.first);
                 }
             }
+            // The unreachable endpoints may be reachable via seed
+            // nodes if their IP addresses have changed. Try them as well.
+            for (auto&& x : _seeds) {
+                if (_unreachable_endpoints.contains(x)) {
+                    // We are already contacting this endpoint
+                    continue;
+                }
+                auto es = get_endpoint_state_for_endpoint_ptr(x);
+                if (es && !is_dead_state(*es)) {
+                    // The endpoint is already live
```

Comment:
is_dead_state is not the way to check if a node is alive. Use gossiper::is_alive.

Reply:
I find it confusing and don't understand the difference. Moreover, if I can reason about endpoint state value, I can not reason about bool _is_alive - who changes it and when. For the purposes of my patch, i.e. being able to contact a node which ip address has changed, both checks will do. I can change it if you like, but as a side note I find endpoint_state::is_alive()  undocumented and ambiguous.

Reply:
Look at the code:
```
bool gossiper::is_dead_state(const endpoint_state& eps) const {
    auto state = get_gossip_status(eps);
    for (auto& deadstate : DEAD_STATES) {
        if (state == deadstate) {
            return true;
        }           
    }               
    return false;   
}  

```

is_dead_state checks the status of a node

gossiper::is_alive checks liveness.

I am not sure why you think both check will do. Checking is_dead_state is definitely wrong.

Reply:
Please take into account that the node which has changed its IP address and is not known in hosts table yet is either missing from endpoint states completely, or is DEAD/UNKNOWN state. It can not be NORMAL or UP or DOWN, because we're not familiar with this node yet.

Reply:
I changed to is_alive() since it doesn't make any difference for me.